server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -2707,7 +2704,9 @@ public void snapshotRemoteStoreIndexShard(
                         store.indexSettings().getUUID(),
                         store.indexSettings().getRemoteStoreRepository(),
                         this.basePath().toString(),
-                        fileNames
+                        fileNames,
+                        pathStrategy.getType(),
+                        pathStrategy.getHashAlgorithm()
```

Comment:
while writing to the file we are passing pathType and hashAlgorithm, however while reading we are using `getRemoteStorePathStrategy` method. would it be better if we directly pass RemoteStorePathStrategy instance in constructor and internally use these attributes while parsing or creating xcontent object?

Reply:
Sure, will do this.

Reply:
I attempted this change, but this has downsides of multiple checks (if/else) conditions to be added and also inhibits the capability to validate path type and hash algorithm against the version. For that reason, I am keeping this unchanged.